diff --git a/modules/session/Content.qml b/modules/session/Content.qml
--- a/modules/session/Content.qml
+++ b/modules/session/Content.qml
@@ -104,6 +104,21 @@ Column {
         Keys.onReturnPressed: exec()
         Keys.onEscapePressed: root.screenState.session = false
         Keys.onPressed: event => {
+            switch (event.key) {
+            case Qt.Key_D:
+                shutdown.exec();
+                event.accepted = true;
+                return;
+            case Qt.Key_L:
+                Quickshell.execDetached(["loginctl", "lock-session"]);
+                event.accepted = true;
+                return;
+            case Qt.Key_R:
+                reboot.exec();
+                event.accepted = true;
+                return;
+            }
+
             if (!Config.session.vimKeybinds)
                 return;
 
